son_slave_thread: trigger measurement request on sta connection

vitalii-komisarenko opened the pull request:
At the moment Associated STA Link Metrics collection is triggered by controller.
It is a vendor-specific message and it is sent just after STA connection.

Vendor-specific approach has disadvantage of not supporting non prplMesh contollers
and it is just redundant since there is a non-VS message for doing similar work.

We would like to replace controller-agent communication with non-VS messages,
but we would like to do it in tiny steps in order to not break anything.

This commit triggers statistics collection on association without
removing any code. Old flows need to be removed/refactored in future commits.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

tomereli commented on agent/src/beerocks/slave/son_slave_thread.cpp:
```diff
@@ -394,6 +394,45 @@ bool slave_thread::handle_cmdu_control_ieee1905_1_message(Socket *sd,
+bool slave_thread::start_monitoring(const beerocks_message::sClientMonitoringParams &params,
+                                    std::shared_ptr<beerocks_header> beerocks_header)
+{
+    LOG(DEBUG) << __func__;
```
This is not very useful - we have a more comprehensive debug log below.
```suggestion
```

tomereli commented on agent/src/beerocks/slave/son_slave_thread.cpp:
```diff
@@ -394,6 +394,45 @@ bool slave_thread::handle_cmdu_control_ieee1905_1_message(Socket *sd,
 }
 
+bool slave_thread::start_monitoring(const beerocks_message::sClientMonitoringParams &params,
+                                    std::shared_ptr<beerocks_header> beerocks_header)
```
It is better to pass `beerocks_header` as reference. It is also in our [CODINGSTYLE.md](https://github.com/prplfoundation/prplMesh/blob/master/CODINGSTYLE.md).

tomereli commented on agent/src/beerocks/slave/son_slave_thread.cpp:
```diff
@@ -394,6 +394,45 @@ bool slave_thread::handle_cmdu_control_ieee1905_1_message(Socket *sd,
+
+    LOG(DEBUG) << "START_MONITORING_REQUEST: mac=" << client_mac << " ip=" << client_ip
+               << " bridge_4addr_mac=" << client_bridge_4addr_mac
+               << " channel=" << std::to_string(params.channel)
```
If you're printing channel, better also print the bandwidth. As a matter of fact, now it makes more sense to also print the operating class if its available in this context.

tomereli commented on agent/src/beerocks/slave/son_slave_thread.cpp:
```diff
@@ -394,6 +394,45 @@ bool slave_thread::handle_cmdu_control_ieee1905_1_message(Socket *sd,
     return true;
 }
 
+bool slave_thread::start_monitoring(const beerocks_message::sClientMonitoringParams &params,
```
It is great that you split the start monitoring to a separate function. However, this should be done in a separate commit.

tomereli commented on agent/src/beerocks/slave/son_slave_thread.cpp:
```diff
@@ -394,6 +394,45 @@ bool slave_thread::handle_cmdu_control_ieee1905_1_message(Socket *sd,
+    auto request_out = message_com::create_vs_message<
+        beerocks_message::cACTION_MONITOR_CLIENT_START_MONITORING_REQUEST>(cmdu_tx,
+                                                                           beerocks_header->id());
+    if (request_out == nullptr) {
```
```suggestion
    if (!request_out) {
```

tomereli commented on agent/src/beerocks/slave/son_slave_thread.cpp:
```diff
@@ -2147,7 +2157,17 @@ bool slave_thread::handle_cmdu_ap_manager_message(Socket *sd,
         send_cmdu_to_controller(cmdu_tx);
 
-        break;
+        beerocks_message::sClientMonitoringParams monitoring_params;
```
Since you made start_monitoring a class method, all this can and should be done inside the function, to keep this switch case even shorter.

tomereli commented on agent/src/beerocks/slave/son_slave_thread.cpp:
```diff
@@ -2147,7 +2157,17 @@ bool slave_thread::handle_cmdu_ap_manager_message(Socket *sd,
+        monitoring_params.bridge_4addr_mac =
+            network_utils::mac_from_string(backhaul_params.bridge_mac);
+        monitoring_params.ipv4    = network_utils::ipv4_from_string(network_utils::ZERO_IP_STRING);
+        monitoring_params.channel = backhaul_params.backhaul_channel; // TODO: unused?
```
If it is indeed unused, please remove this in a separate commit.

tomereli commented on agent/src/beerocks/slave/son_slave_thread.cpp:
```diff
@@ -2147,7 +2157,17 @@ bool slave_thread::handle_cmdu_ap_manager_message(Socket *sd,
+        monitoring_params.ipv4    = network_utils::ipv4_from_string(network_utils::ZERO_IP_STRING);
+        monitoring_params.channel = backhaul_params.backhaul_channel; // TODO: unused?
+        monitoring_params.vap_id  = notification_in->params().vap_id;
+        monitoring_params.is_ire  = 0;
```
This looks wrong. I don't understand the logic completely (perhaps @morantr or @vitalybu can shed some light), but IRE stands for "Intelligent range extender", which means its a repeater (well, range extender differs from repeater in that it has 3 radios, meaning a dedicated radio for backhaul).

From the looks of it, it looks like the whole IRE VS messaging path here should be removed. It is sending a notification to the ap manager that the connected client is a 4addr client, and this whole path is now dead code as it seems. But this should be removed in a separate PR.

For now, I would just remove the is_ire from here completely (in a separate commit).

vitalii-komisarenko replied:
To be honest, I don't have much confidence in these values.
According to the logs from `./test_flows.py`, `ipv4` and `bridge_4addr_mac` are always zero.
Setting `is_ire` to non-zero value breaks `client_steering_dummy` test.

arnout replied:
That's exactly what Tomer is saying: the is_ire != 0 part should be removed entirely.

vitalii-komisarenko replied:
@tomereli , @arnout  Shall I remove the `ACTION_APMANAGER_CLIENT_IRE_CONNECTED_NOTIFICATION` , the list of IREs (`inserted_ire`) managed by ap manager and all the code that uses the list?

Or just remove `is_ire != 0` path from `ACTION_APMANAGER_CLIENT_ASSOCIATED_NOTIFICATION`?

tomereli commented on agent/src/beerocks/slave/son_slave_thread.cpp:
```diff
@@ -515,37 +554,8 @@ bool slave_thread::handle_cmdu_control_message(Socket *sd,
-        message_com::send_cmdu(monitor_socket, cmdu_tx);
-        break;
+        LOG(DEBUG) << "Call start_monitoring from ACTION_CONTROL_CLIENT_START_MONITORING_REQUEST";
+        return start_monitoring(request_in->params(), beerocks_header);
```
What about stop_monitoring? Should be a separate API for that, and it should be called when a client disassociates, doesn't it?

vitalii-komisarenko committed:
beerocks: HOSTAP_MEASUREMENTS: remove unused `channel` field

The `channel` field is transmitted, but unused in HOSTAP_MEASUREMENTS flow.
No new development is expected for this flow, since it is non-standard
and will be partially replaced by Associated STA Link Metrics flow.

It means that `channel` field and related code is a dead code and should be removed.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

arnout commented on agent/src/beerocks/slave/son_slave_thread.cpp:
```diff
@@ -394,6 +394,42 @@ bool slave_thread::handle_cmdu_control_ieee1905_1_message(Socket *sd,
+    //notify monitor
+    auto request_out = message_com::create_vs_message<
+        beerocks_message::cACTION_MONITOR_CLIENT_START_MONITORING_REQUEST>(cmdu_tx,
+                                                                           beerocks_header.id());
```
This is the only use of beerocks_header, so I don't think it makes much sense to pass the entire header.

For the new call, the id parameter can just be zero.

arnout commented on agent/src/beerocks/slave/son_slave_thread.cpp:
```diff
@@ -394,6 +394,42 @@ bool slave_thread::handle_cmdu_control_ieee1905_1_message(Socket *sd,
+        }
+        request_out->mac() = params.mac;
+        message_com::send_cmdu(ap_manager_socket, cmdu_tx);
+    }
```
This part is only relevant when handling the VS message, not when we trigger it autonomously. So I'd leave this outside of the function.

That also resolves Tomer's question about is_ire.

arnout reviewed (approved):
I believe this PR is good enough to be merged with small changes.

morantr commented on agent/src/beerocks/monitor/monitor_thread.cpp:
```diff
@@ -757,7 +756,6 @@ bool monitor_thread::handle_cmdu(Socket *sd, ieee1905_1::CmduMessageRx &cmdu_rx)
         }
 
         sta_node = mon_db.sta_add(sta_mac, vap_id);
-        sta_node->set_ipv4(sta_ipv4);
```
This commit will cause serious bugs. Please drop this commit.
It is not a good idea to remove code that is being used if you are not 100% sure what it does and how it works.

morantr reviewed (changes requested):
Sorry, but I can't approve it.
First, it would really be nice if there was a DR task on that issue mentioned in the PR description.

Second, the drop of the IP from the message it wrong. The monitor implements an arp sending mechanism that uses this IP which being sent by the controller. So removing it will cause unconnected measurements to fail.

I would implement it differently:
1. The start_monitoring does not need to be sent by the agent. You cant add STA-CONNECT handler to monitor bwl, so the monitor gets this event (and also the disconnect event) directly from bwl, and start/stop monitoring directly (which is basically add/remove the client from mon_db).
2. Add a message that updates the monitor with the IP address. The son_slave receives it from the platform manager so all you need to do is to forward it to the monitor and update the station IP add on mon_db.
3. Remove the start/stop_monitoring message from everywhere.

vitalii-komisarenko committed:
agent: monitor: extract ipv4 setting from start_monitoring_sta

At the moment, start_monitoring_sta gets all required data at the same time.
This flow is driven by VS message from the controller.

We want to avoid VS communication with controller and trigger start_monitoring_sta
by AP-STA-CONNECTED event in BWL. In this case ipv4 parameter will be missing
in the original request and will be set later by other means.

Because of it, setting ipv4 to a monitored node is extracted into a separate subroutine.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalybu commented on agent/src/beerocks/monitor/monitor_thread.h:
```diff
@@ -54,6 +54,18 @@ class monitor_thread : public beerocks::socket_thread {
+    /**
+     * @brief mutex to protect the list of STAs being monitored
+     */
+    std::mutex monitoring_sta_mutex;
```
What's that for?
Monitor runs in a single context..

vitalii-komisarenko replied:
Just a habit to add mutexes. But since it is a single thread, I will remove it.

vitalybu commented on agent/src/beerocks/monitor/monitor_thread.cpp:
```diff
@@ -653,6 +653,70 @@ bool monitor_thread::update_ap_stats()
+        return false;
+    }
+
+    if (!_monitored_sta_set_ipv4(sta_mac, sta_ipv4)) {
```
Why a separate method for setting the IP address?
Can't you just pass it to `_start_monitoring_sta()`?

vitalii-komisarenko replied:
At the moment, I am moving in direction described by https://github.com/prplfoundation/prplMesh/pull/1139#pullrequestreview-398271772 .
BWL will give me `mac` and `vap_id` at the same time, but `ipv4` will arrive a bit later from the slave.

vitalii-komisarenko committed:
agent: monitor: add `al_mac` to monitor_thread

Add `al_mac` member variable that represents bridge mac address
and initialize it in monitor_thread constructor.

Bridge MAC is used for statistics collection for Associated STA Link Metrics (ASLM) messages.
Currently it is supplied by VS message from the controller.
It makes it impossible to use ASLM flow with non-prplMesh controller,
but this flow is covered by certification tests (flow 4.7.7).

This commit is a preparatory commit.
It will supply bridge mac before it is removed from the above-metioned VS messages.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
agent: monitor: use `al_mac` in `update_vaps_in_db`

`update_vaps_in_db` has code that calculates equivalent of `al_mac`.
Replace it is with `al_mac`.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
agent: monitor: use `al_mac` for monitoring STAs

Use `al_mac` member variable for monitoring STAs
instead of `set_bridge_4addr_mac` supplied by `ACTION_MONITOR_CLIENT_START_MONITORING_REQUEST`.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
agent: monitor: extract `start_monitoring_sta` function

Currently monitor_stats rely on VS message from the controller
to start monitoring a STA.
We want to do start it automatically.
For the transition period both approaches will co-exist,
so this code will be accessible from two flows.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
agent: monitor: START_MONITORING: print vap_id as int

Change debug output in ACTION_MONITOR_CLIENT_START_MONITORING_REQUEST
to print vap_id as `int`.
By default it was printed as a character that was often in non-printable range.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
agent: son_slave: notify monitor if STA acquired ipv4 address

Notify monitor thread if a STA obtains an IPv4 address.

It is needed to start monitoring STAs.
The old (START_MONITORING_REQUEST) flow is being replaced with a flow
initiated by `STA_Connected` BWL event.
Unfortunately, it is not possible to get IPv4 this way,
so we need a notification from son_slave.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
beerocks: bwl: add STA_Connected event to mon_wlan_hal

Define new type of mon_wlan_hal event: STA_Connected.

Its purpose is to trigger STA monitoring in monitor.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
beerocks: bwl: add STA_Disonnected event to mon_wlan_hal

Define new type of mon_wlan_hal event: STA_Disconnected.

Its purpose is to stop monitoring particular STA in monitor.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
beerocks: bwl: add sACTION_MONITOR_CLIENT_ASSOCIATED_NOTIFICATION

The purpose of this struct is to carry payload (mac and vap_id)
for STA_Connected event in mon_wlan_hal.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
beerocks: bwl: add sACTION_MONITOR_CLIENT_DISCONNECTED_NOTIFICATION

The purpose of this struct is to carry mac address
for STA_Disconnected event in mon_wlan_hal.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
beerocks: bwl: send STA_Disconnected event to monitor in dummy mode

Send STA_Disconnected event to monitor in dummy mode.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
beerocks: bwl: send STA_Disconnected event to monitor in dwpal mode

Send STA_Disconnected event to monitor in dwpal mode.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
beerocks: bwl: send STA_Connected event to monitor in dwpal mode

Send STA_Connected event to monitor in dwpal mode.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
beerocks: bwl: send STA_Connected event to monitor in netlink mode

Send STA_Connected event to monitor in netlink mode.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
beerocks: bwl: send STA_Connected event to monitor in dummy mode

Send STA_Connected event to monitor in dummy mode.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
beerocks: bwl: send STA_Disconnected event to monitor in netlink mode

Send STA_Disconnected event to monitor in netlink mode.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
agent: monitor: add STA_Connected event handler

STA_Connected event will now trigger STA monitoring.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
agent: monitor: remove support of MONITOR_CLIENT_START_MONITORING

Remove MONITOR_CLIENT_START_MONITORING_REQUEST support.

This message is part of VS flow driven by prplMesh controller.
It is replace with STA_Connected event from the BWL.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
agent: monitor: use helper function to set ipv4 of sta

ACTION_MONITOR_CLIENT_NEW_IP_ADDRESS_NOTIFICATION has code
to set ipv4 address of a monitored STA.

Since the code with the same functionality already exists
as a subroutine, we shall use it

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
agent: monitor: add stop_monitoring_sta function

Added a helper function to stop monitoring STA with particular mac address.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
agent: monitor: use stop_monitoring_sta in start_monitoring_sta

start_monitoring_sta removes sta from the list of monitored sta
before adding it again in order to remove all possible old measurements.

This commit replaces that code with stop_monitoring_sta function.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
agent: monitor: add handler for STA_Disconnected BWL event

Add handler for STA_Disconnected BWL event.
The monitor will stop monitoring the disconnected STA if received this event.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
agent: monitor: remove MONITOR_CLIENT_STOP_MONITORING handler

Remove ACTION_MONITOR_CLIENT_STOP_MONITORING_REQUEST handler
because it is replaced with STA_Disconnected BWL event handler.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
agent: son_slave: remove CONTROL_CLIENT_STOP_MONITORING handler.

Remove ACTION_CONTROL_CLIENT_STOP_MONITORING_REQUEST handler.
Its purpose is to stop STA monitoring in VS-specific way.

Since STA monitoring is controlled now by BWL events,
the old flow can be removed.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
agent: son_slave: stop notifying monitor if sta disconnected

There is no need to notify monitor on STA disassociation
since it already recievs this information from BWL
(STA_Disconnected event).

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
agent: son_slave: remove MONITOR_CLIENT_START_MONITORING_REQUEST

Do not send MONITOR_CLIENT_START_MONITORING_REQUEST to monitor
because this flow has been replaced.
Measurements are triggered by BWL, ip address is forwarded from platform manager.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

arnout commented:
I haven't reviewed yet, however:
- 28 commits is a bit too big to review in one shot. Would it be possible/easy to split into 2-3 PRs?
- We are still missing an update to tests/certification/passing_agent_tests.txt to add 4.7.7 to the passing tests.
- The last commit of the PR should trigger the 4.7.7 tests with the magic string in the commit message.

vitalii-komisarenko committed:
passing_agent_tests: add 4.7.7_ETH_* tests

Add 4.7.7_ETH_* tests to the list of passing tests.

MAP-4.7.7_ETH_FH24G:netgear-rax40
MAP-4.7.7_ETH_FH5GL:netgear-rax40
MAP-4.7.7_ETH_FH5GH:netgear-rax40

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
controller: START_MONITORING: do not set bridge mac

This is a preparation to removal of `bridge_4addr_mac` field from `sClientMonitoringParams`.

It was used to supply bridge mac to STA monitoring subsystem,
but now bridge mac is stored as `al_mac` member of `monitor_thread`.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
controller: START_MONITORING: do not set vap_id

This is a preparation to removal of `vap_id` field from `sClientMonitoringParams`.

It was used to supply VAP ID to STA monitoring subsystem,
but now it is supplied by `STA_Connected` BWL monitor event.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
son_slave: START_MONITORING: remove debug output of ipv4

This is a preparation to removal of `ipv4` field from `sClientMonitoringParams`.

It was used to supply IPv4 address to STA monitoring subsystem,
but now it is supplied by `ACTION_MONITOR_CLIENT_NEW_IP_ADDRESS_NOTIFICATION` from son_slave.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
son_slave: START_MONITORING: remove debug output of bridge mac

This is a preparation to removal of `bridge_4addr_mac` field from `sClientMonitoringParams`.

It was used to supply bridge mac to STA monitoring subsystem,
but now bridge mac is stored as `al_mac` member of `monitor_thread`.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
controller: START_MONITORING: do not set ipv4

This is a preparation to removal of `ipv4` field from `sClientMonitoringParams`.

It was used to supply IPv4 address to STA monitoring subsystem,
but now it is supplied by `ACTION_MONITOR_CLIENT_NEW_IP_ADDRESS_NOTIFICATION` from son_slave.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
beerocks: sClientDisconnectionParams: remove vap_id

Remove `vap_id` field from `sClientDisconnectionParams`
because it is not used anymore.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
beerocks: sClientDisconnectionParams: remove bridge_4addr_mac

Remove `bridge_4addr_mac` field from `sClientDisconnectionParams`
because it is not used anymore.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
beerocks: sClientDisconnectionParams: remove ipv4

Remove `ipv4` field from `sClientDisconnectionParams`
because it is not used anymore.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
controller: association_task: don't wait for START_MONITORING_RESPONSE

ACTION_CONTROL_CLIENT_START_MONITORING_REQUEST was used to trigger monitoring on agent's side.
Agent was supposed to answer with ACTION_CONTROL_CLIENT_START_MONITORING_RESPONSE.

That flow has changed. Monitoring is now triggered by BWL `STA_Connected` event.
No notification to the controller is sent.

The absence of START_MONITORING_RESPONSE broke the state machine of the controller.
This response required to switch to CHECK_11K_BEACON_MEASURE_CAP/REQUEST_RSSI_MEASUREMENT_WAIT
if the status of the response was successful.

This code fixes the state machine by changing it state to CHECK_11K_BEACON_MEASURE_CAP/REQUEST_RSSI_MEASUREMENT_WAIT
right after sending START_MONITORING_REQUEST.
The code assumes that the agent is always able to start measurements because:
* the requirements for the measurements are covered by standard.
* the controller has no control over the agents measurements anyway.

START_MONITORING_REQUEST is not removed since it is used to notify AP Manager if an IRE has connected.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
controller: association_task: send START_MONITORING only for IREs

ACTION_CONTROL_CLIENT_START_MONITORING_REQUEST was used to trigger monitoring on agent's side.

That flow has changed. Monitoring is now triggered by BWL `STA_Connected` event.
The old flow has lost its former meaning.
It is now used only to notify AP Manager if an IRE has connected.

This message has effect only if `is_ire` is set to non-zero value.
Otherwise it is discarded by the son_slave.

This commit changes the controller behaviour to only send START_MONITORING_REQUEST
if the connected station is IRE.

It is a preparation to rename ACTION_CONTROL_CLIENT_START_MONITORING_REQUEST
to ACTION_CONTROL_CLIENT_IRE_CONNECTED_NOTIFICATION and to remove `is_ire` field completely.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
beerocks: rename START_MONITORING_REQUEST to IRE_CONNECTED_NOTIFICATION

Rename ACTION_CONTROL_CLIENT_START_MONITORING_REQUEST to ACTION_CONTROL_CLIENT_IRE_CONNECTED_NOTIFICATION.

In past ACTION_CONTROL_CLIENT_START_MONITORING_REQUEST was used to:
* trigger automatic statistics collection on the monitor
* notify AP manager that an IRE has connected
First usage was replaced by `STA_Connected` BWL event
and the duplicating flow has been removed.

The only thing remaining code is AP manager notification.
This commit renames the message in order to better reflect its purpose.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
beerocks: IRE_CONNECTED_NOTIFICATION: carry only mac address

ACTION_CONTROL_CLIENT_IRE_CONNECTED_NOTIFICATION is what is left of
the old ACTION_CONTROL_CLIENT_START_MONITORING_REQUEST.

Among other things it used `is_ire` flag to inform AP manager that a repeater has connected.

Now we send ACTION_CONTROL_CLIENT_IRE_CONNECTED_NOTIFICATION only for IREs,
so `is_ire` flag is now redundant.

The only piece of information left in the message body is the IRE mac address.

This commit replaces the old message body (sClientMonitoringParams,
consisting only of `is_ire` and `mac`) with a single field for the mac address.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
beerocks: remove ACTION_CONTROL_CLIENT_STOP_MONITORING_REQUEST

Remove ACTION_CONTROL_CLIENT_STOP_MONITORING_REQUEST because it is no longer used.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
beerocks: remove ACTION_MONITOR_CLIENT_START_MONITORING_RESPONSE

Remove ACTION_MONITOR_CLIENT_START_MONITORING_RESPONSE because it is no longer used.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
beerocks: remove sClientMonitoringParams

Remove `sClientMonitoringParams` struct because it is no longer used.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
beerocks: remove ACTION_MONITOR_CLIENT_START_MONITORING_REQUEST

Remove ACTION_MONITOR_CLIENT_START_MONITORING_REQUEST because it is no longer used.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
beerocks: remove ACTION_MONITOR_CLIENT_STOP_MONITORING_REQUEST

Remove ACTION_MONITOR_CLIENT_STOP_MONITORING_REQUEST because it is no longer used.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
beerocks: remove ACTION_CONTROL_CLIENT_START_MONITORING_RESPONSE

Remove ACTION_CONTROL_CLIENT_START_MONITORING_RESPONSE because it is no longer used.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
son_slave: do not serve ACTION_MONITOR_CLIENT_START_MONITORING_RESPONSE

Remove ACTION_MONITOR_CLIENT_START_MONITORING_RESPONSE handler because
*_START_MONITORING_RESPONSE flow is no longer used.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
master: son_actions: do not send STOP_MONITORING_REQUEST

Do not send ACTION_CONTROL_CLIENT_STOP_MONITORING_REQUEST
because its purpose was taken over by `STA_Disconnected` BWL event.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
son_master_thread: remove placeholder for START_MONITORING_RESPONSE

Remove placeholder of ACTION_CONTROL_CLIENT_START_MONITORING_RESPONSE handler
because this message is no longer used.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko commented:
Replaced with #1290, #1289 and #1291.
